radicale/storage.py

```diff
@@ -304,23 +357,40 @@ def get(self, href):
         raise NotImplementedError
 
     def get_multi(self, hrefs):
-        """Fetch multiple items. Duplicate hrefs must be ignored.
+        """Fetch multiple items.
+
+        Functionally similar to ``get``, but might bring performance benefits
+        on some storages when used cleverly. It's not required to return the
+        requested items in the correct order. Duplicated hrefs can be ignored.
+
+        Returns tuples with the href and the item or None if the item doesn't
+        exist.
+
+        """
+        return map(lambda href: (href, self.get(href)), hrefs)
```

Comment:
I'm curios about this. AFAIK (and I'm quite sure about this) generator expressions (using parents `(…)`) are like iterators. Opposed to list comprehension (using brackets `[…]`) which are like lists. And iterators are not loading all data into memory at once. But I may be wrong and am really curious to learn why.